FrontEnd/src/components/ProfileDetail/MainInfo/TitleInfo.jsx

```diff
@@ -133,15 +139,30 @@ function TitleInfo({ isAuthorized, data }) {
         </div>
         <div className={classes['title-block__company_region']}>{profile.region}</div>
       </div>
-      {isAuthorized && !ownProfile ? (
-        <button
-          onClick={handleClick}
-          type="button"
-          className={`${classes['title-block__button']} ${isSaved && classes['added_to_saved__button']}`}
-        >
-            <span className={`${classes['title-block__button--text']} ${isSaved && classes['added_to_saved__button--text']}`}>{!isSaved ? 'Додати в збережені' : 'Додано в збережені'}</span>
-          {isAuthorized ? (isSaved ? filledStar : outlinedStar) : null}
-        </button>
+      {isAuthorized ? (
+        <>
+          {!ownProfile && (
+            <button
+              onClick={handleClick}
+              type="button"
+              className={`${classes['title-block__button']} ${isSaved && classes['added_to_saved__button']}`}
+            >
+              <span className={`${classes['title-block__button--text']} ${isSaved && classes['added_to_saved__button--text']}`}>
+                {!isSaved ? 'Додати в збережені' : 'Додано в збережені'}
+              </span>
+              {isAuthorized ? (isSaved ? filledStar : outlinedStar) : null}
```

Comment:
Could we move it into function? double ternary is not the best practice

Reply:
Done. But I doubt I've made it correctly.